packages/prosemirror-resizable-view/src/prosemirror-resizable-view.ts

```diff
@@ -1,172 +1,278 @@
-      handle.dataset.mouseover = '';
-      setHandleDisplay();
+      handleRight.setHandleDisplay();
+      handleLeft.setHandleDisplay();
     });
 
-    handle.addEventListener('mousedown', (e: MouseEvent) => {
-      e.preventDefault();
+    handleRight.dom.addEventListener('mousedown', (e) =>
+      this.resizeHandler(e, view, getPos, handleRight),
+    );
+    handleLeft.dom.addEventListener('mousedown', (e) =>
+      this.resizeHandler(e, view, getPos, handleLeft),
+    );
+
+    outer.append(handleRight.dom);
+    outer.append(handleLeft.dom);
+
+    /**
+     * Only initialize `handleBottom`, `handleBottomRight`, and `handleBottomLeft` in
+     * `ResizableRatioType.Flexible` mode
+     */
+    if (aspectRatio === ResizableRatioType.Flexible) {
+      const handleBottom = new ResizableHandle(ResizableHandleType.Bottom);
+      const handleBottomRight = new ResizableHandle(ResizableHandleType.BottomRight);
+      const handleBottomLeft = new ResizableHandle(ResizableHandleType.BottomLeft);
+
+      outer.addEventListener('mouseover', () => {
+        handleBottom.setHandleDisplay(true);
+        handleBottomRight.setHandleDisplay(true);
+        handleBottomLeft.setHandleDisplay(true);
+      });
 
-      handle.dataset.dragging = 'true';
+      outer.addEventListener('mouseout', () => {
+        handleBottom.setHandleDisplay();
+        handleBottomRight.setHandleDisplay();
+        handleBottomLeft.setHandleDisplay();
+      });
 
-      const startX = e.pageX;
+      handleBottom.dom.addEventListener('mousedown', (e) =>
+        this.resizeHandler(e, view, getPos, handleBottom),
+      );
 
-      const startWidth = getWidthFromNode(this.node) || getSizeFromDom(inner)[0];
+      handleBottomRight.dom.addEventListener('mousedown', (e) =>
+        this.resizeHandler(e, view, getPos, handleBottomRight),
+      );
 
-      const onMouseMove = throttle(100, false, (e: MouseEvent) => {
-        const currentX = e.pageX;
+      handleBottomLeft.dom.addEventListener('mousedown', (e) =>
+        this.resizeHandler(e, view, getPos, handleBottomLeft),
+      );
+      outer.append(handleBottom.dom);
+      outer.append(handleBottomRight.dom);
+      outer.append(handleBottomLeft.dom);
+    }
 
-        const diffX = currentX - startX;
-        outer.style.width = `${startWidth + diffX}px`;
-      });
+    outer.append(element);
 
-      const onMouseUp = (e: MouseEvent) => {
-        e.preventDefault();
+    this.dom = outer;
+    this.#node = node;
+    this.#element = element;
+    this.aspectRatio = aspectRatio;
+  }
 
-        handle.dataset.dragging = '';
-        setHandleDisplay();
+  /**
+   * `createElement` - a method to produce the element DOM element for this prosemirror node.
+   * The subclasses have to implement this abstract method.
+   */
+  abstract createElement(props: {
+    node: ProsemirrorNode;
+    view: EditorView;
+    getPos: () => number;
+    options?: OptionShape;
+  }): HTMLElement;
```

Comment:
@whawker I would like to get some advice from a senior developer. We have two API designs here and I don't know how to choose. Which one is better in your mind?

```ts
class ResizableNodeView {
  constructor(xx, xx, xx, createElement) {
    this.createElement = createElement
  }
}
```

vs 


```ts
abstract class ResizableNodeView {
  constructor(xx, xx, xx) {
  }

  abstract createElement()
}
```

Reply:
@whawker I would like to get some advice from a senior developer. We have two API designs here and I don't know how to choose. Which one is better in your mind?

```ts
class ResizableNodeView {
  constructor(xx, xx, xx, createElement) {
    this.createElement = createElement
  }
}
```

vs 


```ts
abstract class ResizableNodeView {
  constructor(xx, xx, xx) {
  }

  abstract createElement()
}
```

Reply:
I'm my opinion, the second is much more extendable.

The first adds more and more arguments to the constructor, only some of which may be required.

The second could add further abstract methods, - as an aside we could avoid `abstract` altogether? Perhaps we define a default implementation that can be overridden when extended.

Reply:
A third option is something like the `Positioner.fromPositioner` pattern? But that might not be relevant here?

P.s. @ifiokjr said he was planning on reimplementing node views?

Reply:
There is not breaking change for node view so we can implement the feature with the old way for now.

Reply:
@whawker  I can't think of any child node to be implemented here as default node ,that make sense.
Because I think it is designed as a wrapper container,  that has to be extended with thoese node views who want to be resizable.